src/pages/ReimbursementAccount/PersonalInfo/substeps/Address.tsx

```diff
@@ -33,7 +34,8 @@ const INPUT_KEYS = {
     zipCode: personalInfoStepKey.ZIP_CODE,
 };
 
-const REQUIRED_FIELDS = [personalInfoStepKey.STREET, personalInfoStepKey.CITY, personalInfoStepKey.STATE, personalInfoStepKey.ZIP_CODE];
+const STEP_FIELDS = [personalInfoStepKey.STREET, personalInfoStepKey.CITY, personalInfoStepKey.STATE, personalInfoStepKey.ZIP_CODE];
+const REQUIRED_FIELDS = STEP_FIELDS;
```

Comment:
I don't think it's necessary to change the naming and then create the new variable just to satisfy previous order. `fieldIds` is generic enough of a name to take in `REQUIRED_FIELDS` imo

Reply:
I want to clarify, what field keys we have in a particular step. For now, they are the same as the required fields, but can be different. I just want to show, that they can be different in case if we need more fields in the future.

Reply:
I've always been told that premature optimisation is not a good thing. As of now this is just a new variable without an actual reason other than naming

Reply:
OK, if You go with this:
```
const handleSubmit = useReimbursementAccountStepFormSubmit({
        formId: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
        fieldIds: REQUIRED_FIELDS,
        onNext,
        isEditing,
    });
```
someone can ask: why only required fields? That 's why I want to avoid confusions and used two variables

Reply:
Purpose of two variables is to show that data have other purposes.
```
const STEP_FIELDS = [personalInfoStepKey.STREET, personalInfoStepKey.CITY, personalInfoStepKey.STATE, personalInfoStepKey.ZIP_CODE];
const REQUIRED_FIELDS = personalInfoStepKey.STREET, personalInfoStepKey.CITY, personalInfoStepKey.STATE, personalInfoStepKey.ZIP_CODE];
```
is better for You

Reply:
and where is the other purpose as of now?

Reply:
In the hook, I need all fields, but in the validation for function, I need only required fields

Reply:
so in 4 components that you have changed, all of the required fields are also `all fields`... same goes for other steps apparently so what are you trying to optimise for here? I honestly don't see the point

Reply:
I want to clarify that the validation function and hook options can use different fields. 
```
const handleSubmit = useReimbursementAccountStepFormSubmit({
        formId: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
        fieldIds: REQUIRED_FIELDS,
        onNext,
        isEditing,
    });
```
This one can suggest that hook needs required fields.
What do You propose to avoid that?

Reply:
well even if it can - it doesn't as evident by all the steps we have so this is not optimising anything realistically speaking.

>This one can suggest that hook needs required fields.

it doesn't suggest that actually because you would infer what is required for the hook based on the name of the key and not the value ie `fieldIds: BANANAS` would not really mean it needs bananas, right? :D

maybe change the key then to say `allFormFieldIds` or something along those lines? wdyt?